### PR TITLE
Ignore filter on non-list query

### PR DESCRIPTION
**Pull request by amitaibu**
@mateu-aguilo-bosch When there's more comments than code, it shows the solution is a hack. But I didn't think of another way to do it that wouldn't be more hackish.

Got better ideas?

### CODE REVIEW

File: tests/RestfulListTestCase.test [hunk 1]
```diff
@@ -258,6 +258,13 @@ class RestfulListTestCase extends DrupalWebTestCase {
     catch (\Exception $e) {
       $this->fail('Incorrect exception was thrown on invalid key for filter.');
     }
+
+    // Assert filtering doesn't apply for non-list request
+    // (e.g. /api/v1/articles/1?filter[label]=foo), as this might be called from
+    // a formatter plugin, after RESTful's error handling has finished.
+    $query = array('filter' => array('invalid-key' => 'foo'));
+    $result = $this->httpRequest('api/v1/articles/', \RestfulInterface::GET, $query);
```
Comment (e0ipso) [inline tests/RestfulListTestCase.test]: Should this be changed to:
`$result = $this->httpRequest('api/v1/articles/1', \RestfulInterface::GET, $query);`?

**Comment by e0ipso** [comment]
This is one of those things that it can be considered a bug and a feature. You ask for an entity of an ID and matching some filters: then there is one or none. That would be OK too.

Anyways, if we want to pursue this further we need to make sure we strip the filter from the `$request` otherwise the cache layer will have an extra useless bit in the `cid` that will split the cache unnecessarily.

**Comment by amitaibu** [comment]
> This is one of those things that it can be considered a bug and a feature.

Indeed, what I'm fixing over here, is that the exception wasn't caught -- so the response wasn't in JSON.

If the client send wrong data (like I did by mistake), it's fine if the cache isn't optimized.

File: tests/RestfulListTestCase.test (hunk 1 above)
Comment (amitaibu) [inline tests/RestfulListTestCase.test]: correct, I'll fix :)

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**